### DISCUSSION TITLE
Is it possible to use {eval} inside latex blocks?

### DISCUSSION THREAD

**Discussion by germanesosa** [executablebooks/meta#1103] [Q&A]
Hi!

I've been trying to use {eval} inside latex ($...$ or $$ ... $$) in jupyterlab without success (even escaping {} characters).

Is it a missing feature or am I doing something wrong?

Thanks.

**Comment by agoose77** [comment]
At present, this is not possible; LaTeX is treated like a black-box and parsed by a separate parser (e.g. the JS in your browser). This would be fairly hard to implement; the `{eval}` syntax overlaps with some of the LaTeX grammar. As such, I don't expect us to attempt this any time soon.

Is it possible to lift your use of `{eval}` up a level? i.e., rather than injecting content _inside_ LaTeX, you generate the LaTeX itself e.g. in IPython:
```python
from IPython.display import Latex

content_to_insert = "hello world"

display(
    Latex(rf"This is some \LaTeX {content_to_insert}")
)

```
